src/Arcus.Observability.Correlation/ICorrelationInfoAccessorT.cs

```diff
@@ -0,0 +1,14 @@
+﻿namespace Arcus.Observability.Correlation
+{
+    /// <summary>
+    /// Represents a contract to access the <typeparamref name="TCorrelationInfo"/> model.
+    /// </summary>
+    public interface ICorrelationInfoAccessor<TCorrelationInfo> 
+        where TCorrelationInfo : CorrelationInfo
+    {
+        /// <summary>
+        /// Gets or sets the current correlation information initialized in this context.
+        /// </summary>
+        TCorrelationInfo CorrelationInfo { get; set; }
```

Comment:
We should only support get maybe or do you want to use this to automatically set all the required bits, for example in our correlationd middleware?

Maybe less relevant?

Reply:
I thought about it, but the messaging repo is gonna need some setting of the correlation on some way.
This get/set seems like a solution.